incuna_auth/middleware/utils.py

```diff
@@ -0,0 +1,85 @@
+from django.contrib import messages
+from django.http import HttpResponseForbidden, HttpResponseRedirect
+from django.utils.translation import ugettext_lazy as _
+
+from ..models import AccessStateExtensionMixin as AccessState
+
+
+class MiddlewareMixin:
```

Comment:
Hm, you're right.  I don't think it's a base class though (it doesn't have `process_request`, after all :P) and bits of it could certainly be used for non-permission-related middlewares (`is_url_protected` is kind of handy, although the "protected" terminology doesn't suggest it has applications beyond permission).

Struggling to think of further ideas in light of all that, though. Naming is hard :P